### ISSUE TITLE
Unit tests fail with "this Teleport cluster is not licensed for application access"

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#24105]
In the past couple of days we've started getting periodic unit test failures from lib/web/apiserver_test.go tests with a strange error:

```
    apiserver_test.go:4785: 
        	Error Trace:	/__w/teleport/teleport/lib/web/apiserver_test.go:4785
        	Error:      	Received unexpected error:
        	            	this Teleport cluster is not licensed for application access, please contact the cluster administrator
        	Test:       	TestCreateAppSessionHealthCheckAppServer/request_to_application_that_can_be_served
--- FAIL: TestCreateAppSessionHealthCheckAppServer/request_to_application_that_can_be_served (0.21s)
```

I suspect https://github.com/gravitational/teleport/pull/23858 may have been the culprit as it updates the test modules features which probably races with other tests:

https://github.com/gravitational/teleport/commit/bdee053ca985c7a26c5bf2f0d8400fa4834c7a98#diff-7ccc0b653c1df2f24c01f64ed2525daa493092ad3c97df16de3c36e7f05e3911R1928

**Comment by zmb3** [comment]
Yeah, `modules.SetTestModules` can't be used in a `t.Parallel()` test. Maybe we should update `SetTestModules` to have it call `t.Setenv` - this would make a more obvious error when used incorrectly.

**Comment by zmb3** [comment]
https://github.com/gravitational/teleport/actions/runs/4619372480/jobs/8168039295
